# Using IMAPTLS causes loss of emails, if more than one email is in the inbox.
# This is because newer Mail::IMAPClient module returns an array reference
# instead of an array on the ->message action.

diff -Naur otrs3-3.1.7+dfsg1.orig/Kernel/System/MailAccount/IMAPTLS.pm otrs3-3.1.7+dfsg1/Kernel/System/MailAccount/IMAPTLS.pm
--- otrs3-3.1.7+dfsg1.orig/Kernel/System/MailAccount/IMAPTLS.pm	2011-11-14 16:27:31.000000000 +0100
+++ otrs3-3.1.7+dfsg1/Kernel/System/MailAccount/IMAPTLS.pm	2012-07-28 14:06:33.293674056 +0200
@@ -132,9 +132,9 @@
     my $IMAPObject = $Connect{IMAPObject};
     $IMAPObject->select($IMAPFolder) or die "Could not select: $@\n";
 
-    my @Messages = $IMAPObject->messages()
+    my $Messages = $IMAPObject->messages()
         || die "Could not retrieve messages : $@\n";
-    my $NumberOfMessages = scalar @Messages;
+    my $NumberOfMessages = scalar @{$Messages};
 
     if ($CMD) {
         print "$AuthType: I found $NumberOfMessages messages on $Param{Login}/$Param{Host}. "
@@ -147,7 +147,7 @@
         }
     }
     else {
-        for my $Messageno (@Messages) {
+        for my $Messageno ( @{$Messages} ) {
 
             # check if reconnect is needed
             $FetchCounter++;
@@ -243,8 +243,7 @@
             Message => "$AuthType: Fetched $FetchCounter email(s) from $Param{Login}/$Param{Host}.",
         );
     }
-    $IMAPObject->expunge_mailbox($IMAPFolder);
-    $IMAPObject->close();
+    $IMAPObject->close;
     if ($CMD) {
         print "$AuthType: Connection to $Param{Host} closed.\n\n";
     }
